apache/superset
# fix(hive): Update _latest_partition_from_df in HiveEngineSpec to work on tables with multiple indexes

### SUMMARY
Fix #14301

**DataFrame.ix** is deprecated so I replaced it with **.iloc**.
Since Hive return partition in form ds={partition name}/ds={partition name} for table with multiple indexes, I also change the logic to make it work for this case.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
BEFORE:
![image](https://user-images.githubusercontent.com/9434559/115708188-430eb980-a3a2-11eb-9452-4fcb189fd9d3.png)

AFTER:
No more error

### TEST PLAN
- Choose a Hive partitioned table by multiple indexes.
- Check if /superset/extra_table_metadata return correct result with latest value of indexes.
* I tested this in Spark Thrift Server but should also be similar in Hive.

### ADDITIONAL INFORMATION
